c7n/resources/glue.py

```diff
@@ -637,3 +642,30 @@ def get_resource_policy(self, r):
             policy = {}
         r[self.policy_annotation] = policy
         return policy
+
+
+@GlueDataCatalog.filter_registry.register('has-statement')
+class HasStatementFilter(PolicyStatementFilter, GlueCatalogCrossAccount):
```

Comment:
rather than multiple subclassing, preference would be to use composition to get to the cross-account implementation. note in some cases there are two passes on cross account permissions, a process override that gets policies for a set, and a get_resource_policy that simply retrieves the annotation.

Reply:
Good call! I had only considered the MRO working out the way I had expected on this one. Addressed in 3eb9c37